farming/logics/traits/farming.rs

```diff
@@ -0,0 +1,70 @@
+pub use crate::traits::{
+    data::Data,
+    getters::FarmingGetters,
+};
+use openbrush::{
+    contracts::ownable::*,
+    modifiers,
+    traits::{
+        AccountId,
+        Balance,
+        Storage,
+    },
+};
+
+pub const ACC_ARSW_PRECISION: u8 = 12;
+pub const ARTHSWAP_ORIGIN_BLOCK: u32 = 1u32;
+pub const BLOCK_PER_PERIOD: u32 = 215000u32;
+pub const MAX_PERIOD: u8 = 23u8;
+pub const FIRST_PERIOD_REWERD_SUPPLY: Balance = 151629858171523000000u128;
+
+#[openbrush::trait_definition]
+pub trait Farming: Storage<Data> + Storage<ownable::Data> + FarmingGetters {
```

Comment:
Are we going to discard impls, or it is just transient state where default implementations go directly to trait?
Think it is okay with discarding impls.

Reply:
I want to impl directly in the Trait and not use default impl.
As this contract will not be too long (in terms of code lines)